deployment/scripts/process-dataset-gcs.sh

```
@@ -0,0 +1,222 @@
+    local num_batches=$(( (total_files + BATCH_SIZE - 1) / BATCH_SIZE ))
+    log_info "Will process in $num_batches batches"
+
+    # Process in batches
+    local batch_num=1
+    while IFS= read -r batch_files; do
+        [[ -z "$batch_files" ]] && continue
+
+        log_info "=== Batch $batch_num of $num_batches ==="
+
+        # Download
+        local batch_dir=$(download_batch "$batch_files" "$batch_num")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)